src/lib/features/export-import-toggles/export-import-api-usage.test.ts

```diff
@@ -0,0 +1,182 @@
+    {
+        name: 'variantA',
+        weight: 500,
+        payload: {
+            type: 'string',
+            value: 'payloadA',
+        },
+        overrides: [],
+        stickiness: 'default',
+        weightType: 'variable',
+    },
+    {
+        name: 'variantB',
+        weight: 500,
+        payload: {
+            type: 'string',
+            value: 'payloadB',
+        },
+        overrides: [],
+        stickiness: 'default',
+        weightType: 'variable',
+    },
+];
+const exportedFeature: ImportTogglesSchema['data']['features'][0] = {
+    project: 'old_project',
+    name: 'first_feature',
+};
+const constraints: ImportTogglesSchema['data']['featureStrategies'][0]['constraints'] =
+    [
+        {
+            values: ['conduit'],
+            inverted: false,
+            operator: 'IN',
+            contextName: 'appName',
+            caseInsensitive: false,
+        },
+    ];
+const exportedStrategy: ImportTogglesSchema['data']['featureStrategies'][0] = {
+    featureName: defaultFeature,
+    id: '798cb25a-2abd-47bd-8a95-40ec13472309',
+    name: 'default',
+    parameters: {},
+    constraints,
+};
+
+const tags = [
+    {
+        featureName: defaultFeature,
+        tagType: 'simple',
+        tagValue: 'tag1',
+    },
+    {
+        featureName: defaultFeature,
+        tagType: 'simple',
+        tagValue: 'tag2',
+    },
+    {
+        featureName: defaultFeature,
+        tagType: 'special_tag',
+        tagValue: 'feature_tagged',
+    },
+];
+
+const tagTypes = [
+    { name: 'bestt', description: 'test' },
+    { name: 'special_tag', description: 'this is my special tag' },
+    { name: 'special_tag', description: 'this is my special tag' }, // deliberate duplicate
+];
+
+const defaultImportPayload: ImportTogglesSchema = {
+    data: {
+        features: [exportedFeature],
+        featureStrategies: [exportedStrategy],
+        featureEnvironments: [
+            {
+                enabled: true,
+                environment: 'irrelevant',
+                featureName: defaultFeature,
+                name: defaultFeature,
+                variants,
+            },
+        ],
+        featureTags: tags,
+        tagTypes,
+        contextFields: [],
+        segments: [],
+    },
+    project: DEFAULT_PROJECT,
+    environment: DEFAULT_ENV,
+};
+
+test('reject API imports with admin tokens', async () => {
+    const preHook = (app: any) => {
+        app.use('/api/admin/', async (req, res, next) => {
+            const user = new ApiUser({
+                permissions: ['ADMIN'],
+                environment: '*',
+                type: ApiTokenType.ADMIN,
+                tokenName: 'tokenName',
+                secret: 'secret',
+                projects: ['*'],
+            });
+            req.user = user;
+            next();
+        });
+    };
+
+    const { request, destroy } = await setupAppWithCustomAuth(
+        db.stores,
+        preHook,
+    );
+
+    await request
```

Comment:
usually we follow a different testing pattern:
// async call on one line
const {body} = await request.post(....).expect(400)
// sync assertions below
expect().to...

Reply:
Fixed in f2d0702bc6b07ff7b2bac05bf050e25aac1e61dc